http/client/src/main/java/org/wildfly/security/http/client/mechanism/digest/util/DigestHttpMechanismUtil.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.http.client.mechanism.digest.util;
+
+import org.wildfly.security.digest.WildFlyElytronDigestProvider;
+import org.wildfly.security.http.client.exception.ElytronHttpClientException;
+import org.wildfly.security.http.client.utils.ElytronMessages;
+import org.wildfly.security.mechanism.AuthenticationMechanismException;
+
+import static org.wildfly.security.mechanism._private.ElytronMessages.log;
+
+import static org.wildfly.security.mechanism.digest.DigestUtil.parseResponse;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Elytron client for HTTP authentication
+ *
+ * @author <a href="mailto:kekumar@redhat.com">Keshav Kumar</a>
+ */
+public class DigestHttpMechanismUtil {
+
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String CHALLENGE_PREFIX = "Digest ";
+
+    public static HttpRequest createDigestRequest(URI uri, String userName, String password,String authHeader) throws AuthenticationMechanismException {
+        Map<String,String> authParams = updateAuthParams(authHeader);
+        String realm = authParams.getOrDefault("realm", null);
+        String nonce = authParams.getOrDefault("nonce", null);
+        String opaque = authParams.getOrDefault("opaque", null);
+        String algorithm = authParams.getOrDefault("algorithm", "MD5");
+        String qop = authParams.getOrDefault("qop", null);
+        String uriPath = getUriPath(uri);
+        String cnonce = generateCNonce();
+        String nc = String.format("%08x", Integer.parseInt(authParams.getOrDefault("nc", String.valueOf(0))));
```

Comment:
We already use org.wildfly.security.mechanism.digest.DigestUtil class , so we can take its utilities like convertToHexBytesWithLeftPadding for nc here to be consistent